src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -38967,6 +38970,782 @@ static void F(object? x, object? y)
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t._10").WithLocation(13, 13));
         }
 
+        [Fact]
+        [WorkItem(32338, "https://github.com/dotnet/roslyn/issues/32338")]
+        public void CopyStructUnconstrainedFieldNullability()
+        {
+            var source =
+@"#pragma warning disable 649
+struct S<T>
+{
+    internal T F;
+}
+class Program
+{
+    static void M<T>(S<T> s)
+    {
+        if (s.F == null) return;
+        var t = s;
+        t.F.ToString();
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            // https://github.com/dotnet/roslyn/issues/32338: Should not report a warning.
+            comp.VerifyDiagnostics(
+                // (12,9): warning CS8602: Possible dereference of a null reference.
+                //         t.F.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t.F").WithLocation(12, 9));
+        }
+
+        [Fact]
+        [WorkItem(30731, "https://github.com/dotnet/roslyn/issues/30731")]
+        public void StructField_Default_01()
+        {
+            var source =
+@"#pragma warning disable 649
+struct S<T>
+{
+    internal T F;
+}
+class Program
+{
+    static void F<T, U, V>()
+        where U : class
+        where V : struct
+    {
+        default(S<T>).F/*T:T*/.ToString(); // 1
+        default(S<U>).F/*T:U?*/.ToString(); // 2
+        _ = default(S<V?>).F/*T:V?*/.Value; // 3
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (12,9): warning CS8602: Possible dereference of a null reference.
+                //         default(S<T>).F/*T:T*/.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "default(S<T>).F").WithLocation(12, 9),
+                // (13,9): warning CS8602: Possible dereference of a null reference.
+                //         default(S<U>).F/*T:U?*/.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "default(S<U>).F").WithLocation(13, 9),
+                // (14,13): warning CS8629: Nullable value type may be null.
+                //         _ = default(S<V?>).F/*T:V?*/.Value; // 3
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "default(S<V?>).F/*T:V?*/.Value").WithLocation(14, 13));
+            comp.VerifyTypes();
+        }
+
+        [Fact]
+        [WorkItem(30731, "https://github.com/dotnet/roslyn/issues/30731")]
+        public void StructField_Default_02()
+        {
+            var source =
+@"#pragma warning disable 649
+struct S<T>
+{
+    internal T F;
+}
+class Program
+{
+    static void F<T, U, V>()
+        where U : class
+        where V : struct
+    {
+        new S<T>().F/*T:T*/.ToString(); // 1
+        new S<U>().F/*T:U?*/.ToString(); // 2
```

Comment:
>new S<U>() [](start = 8, length = 10)

Do we have a test covering combination of a default value type constructors with object member initializers?